File: Labs/Lab 5 /Main.java
public boolean isValidBST(TreeNode root) {
        return isValidBST(root,Long.MIN_VALUE,Long.MAX_VALUE);
    }

public boolean isValidBST(TreeNode root,long minVal,long maxVal) {
    if (root == null)
        return true;

    if (root.val <= minVal || root.val >= maxVal)
        return false;
    
    boolean left = isValidBST(root.left,minVal,root.val);
    boolean right = isValidBST(root.right,root.val,maxVal);

    return left && right;
}

File: Labs/Lab 5 /Main.py
def isValidBST(self, root: Optional[TreeNode]) -> bool:
    def validBST(root, minVal=float('-inf'), maxVal=float('inf')):
        if root is None:
            return True

        if not (minVal < root.val < maxVal):
            return False

        left = validBST(root.left , minVal, root.val)
        right = validBST(root.right, root.val, maxVal)

        return left and right
    return validBST(root)
